integration/container/remove_test.go

```diff
@@ -111,3 +117,61 @@ func TestRemoveInvalidContainer(t *testing.T) {
 	err := client.ContainerRemove(ctx, "unknown", types.ContainerRemoveOptions{})
 	testutil.ErrorContains(t, err, "No such container")
 }
+
+func TestRemoveContainerOrphaning(t *testing.T) {
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	img := "test-container-orphaning"
+
+	dockerfile1 := `FROM busybox:latest
+	ENTRYPOINT ["true"]`
+
+	dockerfile2 := `FROM busybox:latest
+	ENTRYPOINT ["true"]
+	MAINTAINER Integration Tests`
+
+	// build first dockerfile
+	imageID1 := buildImageWithDockerfile(t, ctx, client, img, dockerfile1)
```

Comment:
I think we should avoid using `build` in tests outside of the image suite.

1. It's going to be slower to build something that to `load`
2. It means that multiple test suites require build. If we want to provide build as a separate service this unnecessary coupling is going to make it a lot more work.

Can we setup this image structure using `image load` api?  If not,  we should make this a unit test.